TransferMethod/Sources/View/Widget/AbstractWidget.swift

```diff
@@ -80,7 +89,12 @@ class AbstractWidget: UIStackView, UITextFieldDelegate {
     }
 
     func isValid() -> Bool {
-        return !isInvalidEmptyValue() && !isInvalidLength() && !isInvalidRegex()
+        var isValid = true
+        if isInvalidEmptyValue() || isInvalidLength() || isInvalidRegex() {
+            trackError()
+            isValid = false
+        }
+        return isValid
```

Comment:
This will probably be more cleaner, although it is fine the way it is too
`var isValid = isInvalidEmptyValue() || isInvalidLength() || isInvalidRegex()
If !isValid {
	trackError()
}
Return isValid`